api/specs.proto

```diff
@@ -298,6 +298,11 @@ message ContainerSpec {
 	oneof init {
 		bool init_value = 23;
 	}
+
+	// Isolation defines the isolation level for engines supporting multiple isolation modes
```

Comment:
Looks like we need an enum here, according to swarmkit's style.

Reply:
I considered that, but after a bit of thinking, it felt that valid values for isolationMode might ultimately depend on the underlying container runtime. For now only dockerd on Windows support different values, but it might change, and we might even require a non fixed set of values.
For exemple, I foresee that with lcow, we might have at some point values like: `hyperv{<something identifying a specific linuxkit/Ubuntu/whatever version>}`

Additionaly, isolationMode is a config setting specific to the container runtime, and has no effect on swarm scheduling. So I don't think we should put validation logic at the swarmkit level here

Reply:
Please match the existing style.

Reply:
@simonferquel The validation consideration here is a valid point and perhaps we could have made this more flexible, but we found that much of the input on the docker side was weakly validated and lead to challenging errors to diagnose in an orchestration system. By ensuring that input is correct, we can eliminate these errors, at the cost of being very literal.